### ISSUE TITLE
Failed to run Java examples

### ISSUE THREAD

**Issue by mariogintili** [mit-nlp/MITIE#79]
hello guys,

after following the setup specified on the readme for java I rain into the following when running the `run_ner.sh`

```
./run_ner.sh
NerExample.java:6: error: package edu.mit.ll.mitie does not exist
import edu.mit.ll.mitie.*;
^
NerExample.java:101: error: cannot find symbol
            StringVector words,
            ^
  symbol:   class StringVector
  location: class NerExample
NerExample.java:102: error: cannot find symbol
            EntityMention ent
            ^
  symbol:   class EntityMention
  location: class NerExample
NerExample.java:22: error: cannot find symbol
        NamedEntityExtractor ner = new NamedEntityExtractor("../../MITIE-models/english/ner_model.dat");
        ^
  symbol:   class NamedEntityExtractor
  location: class NerExample
NerExample.java:22: error: cannot find symbol
        NamedEntityExtractor ner = new NamedEntityExtractor("../../MITIE-models/english/ner_model.dat");
                                       ^
  symbol:   class NamedEntityExtractor
  location: class NerExample
NerExample.java:25: error: cannot find symbol
        StringVector possibleTags = ner.getPossibleNerTags();
        ^
  symbol:   class StringVector
  location: class NerExample
NerExample.java:31: error: cannot find symbol
        StringVector words = global.tokenize(global.loadEntireFile("../../sample_text.txt"));
        ^
  symbol:   class StringVector
  location: class NerExample
NerExample.java:31: error: cannot find symbol
        StringVector words = global.tokenize(global.loadEntireFile("../../sample_text.txt"));
                                             ^
  symbol:   variable global
  location: class NerExample
NerExample.java:31: error: cannot find symbol
        StringVector words = global.tokenize(global.loadEntireFile("../../sample_text.txt"));
                             ^
  symbol:   variable global
  location: class NerExample
NerExample.java:35: error: cannot find symbol
        EntityMentionVector entities = ner.extractEntities(words);
        ^
  symbol:   class EntityMentionVector
  location: class NerExample
NerExample.java:48: error: cannot find symbol
            EntityMention entity = entities.get(i);
            ^
  symbol:   class EntityMention
  location: class NerExample
NerExample.java:66: error: cannot find symbol
        BinaryRelationDetector relDetector = new BinaryRelationDetector("../../MITIE-models/english/binary_relations/rel_classifier_people.person.place_of_birth.svm");
        ^
  symbol:   class BinaryRelationDetector
  location: class NerExample
NerExample.java:66: error: cannot find symbol
        BinaryRelationDetector relDetector = new BinaryRelationDetector("../../MITIE-models/english/binary_relations/rel_classifier_people.person.place_of_birth.svm");
                                                 ^
  symbol:   class BinaryRelationDetector
  location: class NerExample
13 errors
Error: Could not find or load main class NerExample
```

**Comment by mariogintili** [comment]
additionally here's how the `mitielib/java` folder structure looks like https://gist.github.com/mariogintili/d1cd0da3abaf536a694ed6da0993de7d

I also changed the CLASS_PATH to export the following:

```
export CLASSPATH=../../mitielib/java/build/lib/javamitie.jar:. 
```

**Comment by davisking** [comment]
Did you follow the _**exact**_ instructions?

```
cd mitielib/java
mkdir build
cd build
cmake ..
cmake --build . --config Release --target install
```

**Comment by mariogintili** [comment]
@davisking I have 😄 

I'm using a relatively new OS(Mac OS Sierra) think that could be part of the problem?

**Comment by davisking** [comment]
Is there a jar file in the mitielib folder?  That cmake command runs an
install operation that copies the jar to mitielib.  It sounds like that
didn't happen.

**Comment by mariogintili** [comment]
@davisking Hello,

There are 2

- [one here](https://gist.github.com/mariogintili/d1cd0da3abaf536a694ed6da0993de7d#file-foo-sh-L278)
- [and another one here](https://gist.github.com/mariogintili/d1cd0da3abaf536a694ed6da0993de7d#file-foo-sh-L283)

None of them seem are located where the ENV var `CLASSPATH` points to

**Comment by davisking** [comment]
What happens when you try to run the install target on its own.  Do `make install`.  What happens?  It should copy the jar.

**Comment by mariogintili** [comment]
@davisking not sure I'm following, I should run `make install` from the java dir?

**Comment by davisking** [comment]
From the build folder

**Comment by mariogintili** [comment]
```shell
$ pwd
/Users/mariogintili/dev/MITIE/mitielib/java/build

$ make --debug install
GNU Make 3.81
Copyright (C) 2006  Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.

This program built for i386-apple-darwin11.3.0
Reading makefiles...
Updating goal targets....
 File `install' does not exist.
Must remake target `install'.
make: *** No rule to make target `install'.  Stop.
```

**Comment by davisking** [comment]
There is definitely an install target.  I don't know what's wrong on your
system.  I would delete everything and try getting a fresh copy from github.

**Comment by SakthivelAnand** [comment]
I did all the instructions on ReadME......i got two jar files from the build folder with name as javamitie.jar
..i add these jar files to my java mitie example NER program ...but still it says 


Native code library failed to load. 
java.lang.UnsatisfiedLinkError: no javamitie in java.library.path

i given class path to etc/mitie/..../javamitie.jar

What i want to do..for run this example sucessfully..I have tried on it for more than 3 days ..but still i can't find out what is the problem with these configuration process.
I am working on Ubuntu and Eclipse.
